src/components/eip3643/Admin.tsx

```diff
@@ -49,6 +52,21 @@ export default function Admin() {
     setTokenSelected(tokenItem);
   };
 
+  const readTokensList = () => {
+    const unsubTokens: WatchContractEventReturnType =
+      watchTrexFactoryTrexSuiteDeployedEvent({
+        onLogs: (data) => {
+          setDeployedTokens(((prev: any) => {
+            return [...prev, ...data];
+          }) as any);
+        },
+      });
+
+    setTimeout(() => {
+      unsubTokens();
+    }, 10000);
```

Comment:
> The case we are trying to solve here - refreshing the tokens list faster then our cooldown - is tricky and I feel duplicating the watcher is not the optimal way.
> 
> Let's try to keep only one watcher on the higher level (in EIP3643 page) and try to reload it forcefully. How about having some state variable in the dependencies of that watcher, which we can update here with "refresh" click and that would trigger unsub/sub sequence. This might work, wdyt?

Amazing idea)